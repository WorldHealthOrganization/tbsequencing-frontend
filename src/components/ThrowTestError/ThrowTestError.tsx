/** @jsxImportSource @emotion/react */
import React from 'react';
import BasePage from '../BasePage';
import { AppButton } from '../AppButton/AppButton';

export const ThrowTestError = () => {
  const onErrorButtonClick = () => {
    throw new Error('This is a test error thrown by test error page');
  };

  return (
    <BasePage pageHeader="This is a test error page to test Sentry.io">
      <AppButton
        onClick={onErrorButtonClick}
        variant="contained"
        size="large"
      >
        Press this button to throw JS error
      </AppButton>
    </BasePage>
  );
};

export default ThrowTestError;
