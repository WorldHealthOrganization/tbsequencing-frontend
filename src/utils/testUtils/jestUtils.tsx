import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
