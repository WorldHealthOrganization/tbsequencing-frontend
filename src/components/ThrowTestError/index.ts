import ThrowTestError from './ThrowTestError';

export default ThrowTestError;
