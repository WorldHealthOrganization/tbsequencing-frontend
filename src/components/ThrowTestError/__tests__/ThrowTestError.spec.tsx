import React from 'react';
import { render, getTreeJSONWithRedux } from '../../../utils/testUtils/jestUtils';
import ThrowTestError from '../ThrowTestError';

describe('ThrowTestError component test', () => {
  test('renders ThrowTestError component without error', () => {
    render(
      <ThrowTestError />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <ThrowTestError />,
    );

    expect(tree).toMatchSnapshot();
  });
});
