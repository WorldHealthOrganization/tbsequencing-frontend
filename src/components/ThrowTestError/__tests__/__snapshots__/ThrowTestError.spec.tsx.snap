// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThrowTestError component test Should render correctly 1`] = `
<div
  className="css-xn17ji-BasePage"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-177n4ga-MuiTypography-root-BaseText"
  >
    This is a test error page to test Sentry.io
  </p>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-1sfzpsj-MuiButtonBase-root-MuiButton-root-AppButton"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
    >
      Press this button to throw JS error
    </p>
  </button>
</div>
`;
